reference-implementation/to-upstream-wpts/readable-byte-stream/readable-byte-stream.js

```diff
@@ -0,0 +1,1960 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('../resources/rs-utils.js');
+  self.importScripts('/resources/testharness.js');
+}
+
+test(() => {
+  assert_throws(new TypeError(), () => new ReadableStream().getReader({ mode: 'byob' }));
+}, 'getReader({mode: "byob"}) throws on non-bytes streams');
+
+
+test(() => {
+  // Constructing ReadableStream with an empty underlying byte source object as parameter shouldn't throw.
+  new ReadableStream({ type: 'bytes' });
+}, 'ReadableStream with byte source can be constructed with no errors');
+
+async_test(t => {
+  let startCalled = false;
+  let controller;
+
+  new ReadableStream({
+    start(c) {
+      controller = c;
+      startCalled = true;
+    },
+    pull() {
+      assert_true(startCalled, 'start has been called');
+      assert_equals(controller.desiredSize, 256, 'desiredSize');
+      t.done();
+    },
+    type: 'bytes'
+  }, {
+    highWaterMark: 256
+  });
+
+}, 'ReadableStream with byte source: Construct and expect start and pull being called');
+
+async_test(t => {
+  let pullCount = 0;
+  let checkedNoPull = false;
+
+  let resolveStartPromise;
+
+  new ReadableStream({
+    start() {
+      return new Promise(resolve => {
+        resolveStartPromise = resolve;
+      });
+    },
+    pull() {
+      if (checkedNoPull) {
+        t.done();
+      }
+
+      ++pullCount;
+    },
+    type: 'bytes'
+  }, {
+    highWaterMark: 256
+  });
+
+  Promise.resolve().then(() => {
+    assert_equals(pullCount, 0);
+    checkedNoPull = true;
+    resolveStartPromise();
+  });
+
+}, 'ReadableStream with byte source: No automatic pull call if start doesn\'t finish');
+
+async_test(t => {
+  new ReadableStream({
+    pull() {
+      assert_unreached('pull must not be called');
+      t.done();
+    },
+    type: 'bytes'
+  }, {
+    highWaterMark: 0
+  });
+
+  Promise.resolve().then(() => {
+    t.done();
+  });
+}, 'ReadableStream with byte source: Construct with highWaterMark of 0');
+
+async_test(t => {
+  const stream = new ReadableStream({
+    type: 'bytes'
+  });
+
+  const reader = stream.getReader();
+  reader.releaseLock();
+
+  reader.closed.then(() => {
```

Comment:
Should be a promise_test with promise_rejects. Same for most of the async_tests that follow and end by checking reader.closed.